Homework_Week_4/task_2.py

```diff
@@ -0,0 +1,38 @@
+def cash_withdraw():
+    balance = 100.00
+    print("Your balance is",balance,)
+    try:
+        withdrawal_amount = float(input("How much would you like to withdraw? "))
+        if withdrawal_amount > balance:
+            raise Exception
+            return
+        else:
+            balance = balance - withdrawal_amount
+            print("The remaining balance in your account is $:{:2f}".format(balance))
+    except:
+        print("Insufficient funds to withdraw this amount")
+
+def verify_pin(pin):
+    if pin == '1234':
+        return True
+    else:
+        return False
+
+def log_in():
+    attempt = 0
+    while attempt < 3:
+        pin = input('Please Enter Your 4 Digit Pin: ')
+        if verify_pin(pin):
+            print("Correct pin")
+            cash_withdraw()
+            break
+        else:
+            print("Invalid pin")
+            attempt += 1
+    print("Thank you for using the ATM.")
+
+def main_menu():
+    print("Welcome to the atm!")
+    log_in()
+
+main_menu()
```

Comment:
⚠️ (-4) well done on completing the program. I like your approach to dividing it into multiple functions, great job.
However, to make this better, you should exclude the cash_withdrawal() method from the 'login' method.
As a general principle, one method should only do one thing. In this case your `login` method has too many responsiblity i.e. handle login and it also handle `cash withdrawal`

Take a look on the function design below:

```python
def verify_pin(pin):
   # your logic

def login(pin):
   # your logic, to verify including the retry 3 times

def cash_withdraw(balance, amount): 
   # your logic to withdraw


atm():
   if login() == True
      cash_withdraw()
   else
       exit()

```